app.js

```diff
@@ -102,6 +102,19 @@ socket.on('start', function (data) {
   data[defaultSpan].responses.pop();
   data[defaultSpan].os.pop();
 
+  if(!data[defaultSpan].os[data[defaultSpan].os.length - 1]) {
+    data[defaultSpan].os[data[defaultSpan].os.length - 1]= {
```

Comment:
What if `os` length will be 0? Program would try to insert something at index `-1` ;)